Entity/Repositories/MailgunEventRepository.php

```diff
@@ -126,6 +126,22 @@ public function getRecipients()
         return $result;
     }
 
+    /**
+     * Get last known sender IP based on stored events list
+     * @return string|null
+     */
+    public function getLastKnownSenderIp()
+    {
+        $q = $this->getEntityManager()->createQueryBuilder()
+            ->select('e.ip as ip')
```

Comment:
this is not the "sender ip". This is the server that sent back the MailgunEvent to the app. The sender IP can be found in the message headers of the "delivered" events. E.g.  209.61.151.239 in:
```
    MessageId: 444b15f54fc8c63dd93e8c042b83b168@azine.me
    MessageHeaders:

        Sender: error@azine.me
        X-Mailgun-Sending-Ip: 209.61.151.239
        X-Mailgun-Sid: WyIwN2U4YyIsICJzdXBwb3J0QGF6aW5lLm1lIiwgIjA2MjkzIl0=
        Received: from azine.me (b4.tophost.ch [194.150.248.224]) by mxa.mailgun.org with ESMTP id     5936505f.7f0dfc107db0-smtp-out-n01; Tue, 06 Jun 2017 06:49:03 -0000 (UTC)
        Message-Id: <444b15f54fc8c63dd93e8c042b83b168@azine.me>
        Date: Tue, 06 Jun 2017 08:48:16 +0200
        ....
```